File: Algo/Easy/0088_Merge_Sorted_Array/Solution.py
from typing import *

## 2022-05-04
class Solution:
    def merge(self, nums1: List[int], m: int, nums2: List[int], n: int) -> None:
        """
        Do not return anything, modify nums1 in-place instead.
        """
        arr = []
        idx1 = idx2 = 0
        while idx1 < m and idx2 < n:
            if nums1[idx1] < nums2[idx2]:
                arr.append(nums1[idx1])
                idx1 += 1
            else:
                arr.append(nums2[idx2])
                idx2 += 1
        
        while idx1 < m:
            arr.append(nums1[idx1])
            idx1 += 1
        while idx2 < n:
            arr.append(nums2[idx2])
            idx2 += 1
        
        for i in range(n+m):
            nums1[i] = arr[i]
                    

File: Algo/Easy/0088_Merge_Sorted_Array/Solution.java
import java.util.Arrays;

// 2021-04-29
class Solution {
    public void merge(int[] nums1, int m, int[] nums2, int n) {
            
        int count = 0;
        for (int i = 0; i < nums1.length; i++) {
            if (nums1[i] == 0 && count < n) {
                nums1[i] = nums2[count++];
            }
        }
        Arrays.sort(nums1);
    }
}

// 2021-11-11
class Solution2 {
    public void merge(int[] nums1, int m, int[] nums2, int n) {
        for (int i = m; i < nums1.length; i++) {
            nums1[i] = nums2[i-m];
        }
        Arrays.sort(nums1);
    }
}